azurerm/internal/services/sentinel/sentinel_data_connector_office_365.go

```diff
@@ -0,0 +1,244 @@
+	id := parse.NewDataConnectorID(workspaceId.SubscriptionId, workspaceId.ResourceGroup, workspaceId.WorkspaceName, name)
+
+	if d.IsNewResource() {
+		resp, err := client.Get(ctx, id.ResourceGroup, operationalInsightsResourceProvider, id.WorkspaceName, name)
+		if err != nil {
+			if !utils.ResponseWasNotFound(resp.Response) {
+				return fmt.Errorf("checking for existing Sentinel Data Connector Office 365 %q: %+v", id, err)
+			}
+		}
+
+		id := dataConnectorID(resp.Value)
+		if id != nil && *id != "" {
+			return tf.ImportAsExistsError("azurerm_sentinel_data_connector_office_365", *id)
+		}
+	}
+
+	tenantId := d.Get("tenant_id").(string)
+	if tenantId == "" {
+		tenantId = meta.(*clients.Client).Account.TenantId
+	}
+
+	exchangeEnabled := d.Get("exchange_enabled").(bool)
+	sharePointEnabled := d.Get("share_point_enabled").(bool)
+	teamsEnabled := d.Get("teams_enabled").(bool)
+
+	// Service will not create the DC in case non of the toggle is enabled.
+	if !exchangeEnabled && !sharePointEnabled && !teamsEnabled {
+		return fmt.Errorf("one of `exchange_enabled`, `share_point_enabled` and `teams_enabled` should be `true`")
+	}
+
+	exchangeState := securityinsight.Enabled
+	if !exchangeEnabled {
+		exchangeState = securityinsight.Disabled
+	}
+
+	sharePointState := securityinsight.Enabled
+	if !sharePointEnabled {
+		sharePointState = securityinsight.Disabled
+	}
+
+	teamsState := securityinsight.Enabled
+	if !teamsEnabled {
+		teamsState = securityinsight.Disabled
+	}
+
+	param := securityinsight.OfficeDataConnector{
+		Name: &name,
+		OfficeDataConnectorProperties: &securityinsight.OfficeDataConnectorProperties{
+			TenantID: &tenantId,
+			DataTypes: &securityinsight.OfficeDataConnectorDataTypes{
+				Exchange: &securityinsight.OfficeDataConnectorDataTypesExchange{
+					State: exchangeState,
+				},
+				SharePoint: &securityinsight.OfficeDataConnectorDataTypesSharePoint{
+					State: sharePointState,
+				},
+				Teams: &securityinsight.OfficeDataConnectorDataTypesTeams{
+					State: teamsState,
+				},
+			},
+		},
+		Kind: securityinsight.KindOffice365,
+	}
+
+	// Service avoid concurrent updates of this resource via checking the "etag" to guarantee it is the same value as last Read.
+	if !d.IsNewResource() {
+		resp, err := client.Get(ctx, id.ResourceGroup, operationalInsightsResourceProvider, id.WorkspaceName, name)
+		if err != nil {
+			return fmt.Errorf("retrieving Sentinel Data Connector Office 365 %q: %+v", id, err)
+		}
+
+		if err := assertDataConnectorKind(resp.Value, securityinsight.DataConnectorKindOffice365); err != nil {
+			return fmt.Errorf("asserting Sentinel Data Connector of %q: %+v", id, err)
+		}
+		param.Etag = resp.Value.(securityinsight.OfficeDataConnector).Etag
+	}
+
+	_, err = client.CreateOrUpdate(ctx, id.ResourceGroup, operationalInsightsResourceProvider, id.WorkspaceName, id.Name, param)
+	if err != nil {
+		return fmt.Errorf("creating Sentinel Data Connector Office 365 %q: %+v", id, err)
```

Comment:
we can reduce this to:

```suggestion
		return fmt.Errorf("creating %s: %+v", id, err)
```